pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -113,9 +113,44 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 	if err != nil {
 		return api.Environment{}, err
 	}
+	setProductLabels(cr, &mergedEnv)
 	return mergedEnv, nil
 }
 
+func setProductLabels(cr *api.KieApp, env *api.Environment) {
+	consoleComponent := "Console"
+	if envConstants, found := constants.EnvironmentConstants[cr.Status.Applied.Environment]; found {
+		consoleComponent = envConstants.App.FriendlyName
+	}
+	setObjectLabels(cr, &env.Console, consoleComponent)
+	for index := range env.Servers {
+		setObjectLabels(cr, &env.Servers[index], "KIE Server")
+	}
+	setObjectLabels(cr, &env.SmartRouter, "Smart Router")
+	setObjectLabels(cr, &env.ProcessMigration, "Process Migration")
+}
+
+func setObjectLabels(cr *api.KieApp, object *api.CustomObject, component string) {
+	for index, obj := range object.DeploymentConfigs {
+		object.DeploymentConfigs[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, component)
+	}
+	for index, obj := range object.StatefulSets {
+		object.StatefulSets[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, component)
+	}
+}
+
+func setLabels(cr *api.KieApp, labels map[string]string, component string) map[string]string {
+	if labels == nil {
+		labels = map[string]string{}
+	}
+	labels[constants.LabelRHproductName] = constants.ProductName
+	labels[constants.LabelRHproductVersion] = cr.Status.Applied.Version
+	labels[constants.LabelRHcomponentName] = component
+	labels[constants.LabelRHcomponentVersion] = cr.Status.Applied.Version
+	labels[constants.LabelRHcomponentType] = "application"
```

Comment:
Maybe componente type could be RHDM or RHPAM?
not sure, @ibek @krisv what do you guys think?

Reply:
No, it needs to be application or infrastructure. In our case ( Process automation ) I think that all our components are of application type. We need to align with the Integration team here.